### PR TITLE
Add matchstickworks normiepad

### PR DESCRIPTION
**Pull request by MatchstickWorks**
## Description

Adding a new board to MatchstickWorks keyboards, NormiePad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by MatchstickWorks** [comment]
The lint seems to be failing on a file that I didn't modify. Please advise?

**Comment by waffle87** [comment]
> The lint seems to be failing on a file that I didn't modify. Please advise?

No need to worry about it.

File: keyboards/matchstickworks/normiepad/info.json
Comment (drashna) [inline keyboards/matchstickworks/normiepad/info.json:24]: THis appears to match the numpad_6x4 layout, and the config here should be changed to match: 
https://github.com/qmk/qmk_firmware/tree/master/layouts/default/numpad_6x4

File: keyboards/matchstickworks/normiepad/readme.md
Comment (drashna) [inline keyboards/matchstickworks/normiepad/readme.md:3]: URL 404's.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/normiepad/info.json
Comment (MatchstickWorks) [reply keyboards/matchstickworks/normiepad/info.json:24]: Do you mean just changing "LAYOUT" to "LAYOUT_numpad_6x4" (plus the keymap.c rename), or is there something more? Adding the ASCII graphic to keymap.c? I'm unsure what this ask is specifically.

File: keyboards/matchstickworks/normiepad/readme.md
Comment (MatchstickWorks) [reply keyboards/matchstickworks/normiepad/readme.md]: fixed under 296fe54

File: keyboards/matchstickworks/normiepad/info.json
Comment (drashna) [reply keyboards/matchstickworks/normiepad/info.json:24]: Specifically, to change the name, tweak the layout to match the community layout, and: 
```json
    "community_layouts": [ "numpad_6x4" ],
```

Comment (MatchstickWorks) [reply keyboards/matchstickworks/normiepad/info.json:24]: Mine appears to be wired up a little differently than the default 6x4. The `Plus` and `Enter` are on a row higher than the default, won't that cause issues?

I did notice a mistake with my `h` and `w` tags on info.json being wrong (copied over from my other design) so at least I'll be able to fix that.

File: keyboards/matchstickworks/normiepad/keymaps/default/keymap.c
Comment (dunk2k) [inline keyboards/matchstickworks/normiepad/keymaps/default/keymap.c:17]:
```suggestion
};

```

File: keyboards/matchstickworks/normiepad/keymaps/via/keymap.c
Comment (dunk2k) [inline keyboards/matchstickworks/normiepad/keymaps/via/keymap.c:17]:
```suggestion
};

```

**Review by dunk2k — CHANGES_REQUESTED** [review]
add EoF where missing

File: keyboards/matchstickworks/normiepad/info.json
Comment (MatchstickWorks) [reply keyboards/matchstickworks/normiepad/info.json:24]: @drashna Bumping my question

Comment (drashna) [reply keyboards/matchstickworks/normiepad/info.json:24]: Sorry.  Yes and no.  You'd want to move the positions in the info.json to match the community layout.  However, how they are wired up doesn't actually matter to the info.json.  As long as the correct matrix positions are assigned, it will work correctly.   You'd need to alter the keymap.c files, though. 

But my point is that you should use the community layout, whenever possible.  It allows users to maintain one keymap that is usable on different hardware and makes setting things up for them easier.

File: keyboards/matchstickworks/normiepad/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/matchstickworks/normiepad/keymaps/default/keymap.c:14]:
```suggestion
        KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
        KC_P0,            KC_PDOT, KC_PENT
```

File: keyboards/matchstickworks/normiepad/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/matchstickworks/normiepad/keymaps/via/keymap.c:14]:
```suggestion
        KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
        KC_P0,            KC_PDOT, KC_PENT
```

File: keyboards/matchstickworks/normiepad/info.json
Comment (drashna) [inline keyboards/matchstickworks/normiepad/info.json:46]:
```suggestion
                {"label": "Bksp", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "=", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "(", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": ")", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "Num Lock", "matrix": [1, 0], "x": 0, "y": 1},
                {"label": "/", "matrix": [1, 1], "x": 1, "y": 1},
                {"label": "*", "matrix": [1, 2], "x": 2, "y": 1},
                {"label": "-", "matrix": [1, 3], "x": 3, "y": 1},
                {"label": "7", "matrix": [2, 0], "x": 0, "y": 2},
                {"label": "8", "matrix": [2, 1], "x": 1, "y": 2},
                {"label": "9", "matrix": [2, 2], "x": 2, "y": 2},
                {"label": "4", "matrix": [3, 0], "x": 1, "y": 3},
                {"label": "5", "matrix": [3, 1], "x": 2, "y": 3},
                {"label": "6", "matrix": [3, 2], "x": 3, "y": 3},
                {"label": "+", "matrix": [2, 3], "x": 3, "y": 2, "h": 2},
                {"label": "1", "matrix": [4, 0], "x": 0, "y": 4},
                {"label": "2", "matrix": [4, 1], "x": 1, "y": 4},
                {"label": "3", "matrix": [4, 2], "x": 2, "y": 4},
                {"label": "0", "matrix": [5, 0], "x": 0, "y": 5, "w": 2},
                {"label": ".", "matrix": [5, 2], "x": 2, "y": 5},
                {"label": "Enter", "matrix": [4, 3], "x": 3, "y": 4, "h": 2},
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/normiepad/info.json
Comment (MatchstickWorks) [reply keyboards/matchstickworks/normiepad/info.json:24]: Ah, okay! Thanks for the explainer! I'll keep that in mind for future boards.

**Review by waffle87 — APPROVED** [review]

**Comment by MatchstickWorks** [comment]
Hey @drashna Could I get a re-review on this? I think the PR is still thinking a requested chance isn't resolved, but I believe we fixed it further down the discussion line.

**Comment by MatchstickWorks** [comment]
I am still waiting for a review from @drashna 

It's unclear to me what still needs to be addressed with this PR.

File: keyboards/matchstickworks/normiepad/info.json
Comment (fauxpark) [inline keyboards/matchstickworks/normiepad/info.json:1]: This file should be renamed to `keyboard.json`.

File: keyboards/matchstickworks/normiepad/rules.mk
Comment (fauxpark) [inline keyboards/matchstickworks/normiepad/rules.mk:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/normiepad/keyboard.json
Comment (fauxpark) [inline keyboards/matchstickworks/normiepad/keyboard.json:47]:
```suggestion
                {"label": "Enter", "matrix": [4, 3], "x": 3, "y": 4, "h": 2}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**
